### PR TITLE
Refactors radiation

### LINKED ISSUE #2052: Radiation pulses does no damage

### PR DESCRIPTION
**Pull request by Zumorica**
Fixes #2052
Now you can create radiation pulses easily! (without fancy graphics sadly)

- Future refactoring: Make radiation pulses NOT rely on entities.

### CODE REVIEW

**Comment by Zumorica** [comment]
CC @metalgearsloth for radiation changes and CC @DrSmugleaf for bodysystem changes

File: Content.Server/GameObjects/Components/StationEvents/RadiationPulseComponent.cs [hunk 1]
```diff
@@ -18,48 +18,68 @@ public sealed class RadiationPulseComponent : SharedRadiationPulseComponent
...
 
         public override ComponentState GetComponentState()
         {
-            return new RadiationPulseMessage(_endTime);
+            return new RadiationPulseState(_endTime, Range);
+        }
+
+        public void Update(float frameTime)
+        {
+            if (!Decay || !Owner.Deleted)
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/StationEvents/RadiationPulseComponent.cs:76]: Should this be Owner.Deleted or am I misreading it

File: Content.Server/GameObjects/Components/Body/BodyManagerComponent.cs [hunk 2]
```diff
@@ -937,6 +940,13 @@ private static float DistanceToNearestFoot(BodyManagerComponent body, IBodyPart
         }
 
         #endregion
+
+        public void RadiationAct(float frameTime, RadiationPulseComponent radiation)
+        {
+            var totalDamage = Math.Max((int)(frameTime * radiation.DPS), 1);
+
+            ChangeDamage(DamageType.Radiation, totalDamage, false, radiation.Owner);
+        }
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/Body/BodyManagerComponent.cs:949]: Putting radiation on the bodymanager feels weird to me but I don't know bodysystem well enough to comment further.

**Review by metalgearsloth — COMMENTED** [review]
RadiationPulse stuff looks good to me.

File: Content.Server/GameObjects/Components/StationEvents/RadiationPulseComponent.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Server/GameObjects/Components/StationEvents/RadiationPulseComponent.cs]: You're right, fixed now

File: Content.Server/GameObjects/Components/Body/BodyManagerComponent.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Server/GameObjects/Components/Body/BodyManagerComponent.cs:949]: Let's wait for Smug to comment on this, then :relieved:

Comment (DrSmugleaf) [reply Content.Server/GameObjects/Components/Body/BodyManagerComponent.cs:949]: It is weird indeed unless you want only entities with a body to be able to be damaged by radiation. If that's the case though that would be better achieved with a radiation immune ResistanceSet, so I'm in favor of moving this to DamageableComponent as its not even guaranted that all mobs will have a body manager component anyways.

File: Content.Client/GameObjects/Components/StationEvents/RadiationPulseComponent.cs [hunk 3]
```diff
@@ -9,31 +9,17 @@ namespace Content.Client.GameObjects.Components.StationEvents
...
-
-        public override float Range
-        {
-            get => _range;
-            set
-            {
-                _range = value;
-                Dirty();
-            }
-        }
```
Comment (DrSmugleaf) [inline Content.Client/GameObjects/Components/StationEvents/RadiationPulseComponent.cs]: Removing this means that changing the values on the client may potentially put the radiation pulse out of sync with the server.

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

File: Content.Client/GameObjects/Components/StationEvents/RadiationPulseComponent.cs (hunk 3 above)
Comment (Zumorica) [reply Content.Client/GameObjects/Components/StationEvents/RadiationPulseComponent.cs]: You shouldn't be able to modify these values on the client at all

**Pull request merged**